src/components/new-safe/steps/Step2/index.tsx

```diff
@@ -49,11 +49,18 @@ const CreateSafeStep2 = ({
     },
   })
 
-  const { register, handleSubmit, control, watch, formState, getValues } = formMethods
+  const { handleSubmit, control, watch, formState, getValues, setValue } = formMethods
 
   const threshold = watch(CreateSafeStep2Fields.threshold)
 
-  const { fields: ownerFields, append: appendOwner, remove: removeOwner } = useFieldArray({ control, name: 'owners' })
+  const { fields: ownerFields, append: appendOwner, remove } = useFieldArray({ control, name: 'owners' })
+
+  const removeOwner = (index: number): void => {
+    if (ownerFields.length === threshold) {
+      setValue('threshold', threshold - 1)
```

Comment:
Its possible to do `setValue(CreateSafeStep2Fields.threshold, threshold - 1)` if we set the `CreateSafeStep2Form` type differently further up:

```
export type CreateSafeStep2Form = {
  [CreateSafeStep2Fields.owners]: NamedAddress[]
  [CreateSafeStep2Fields.threshold]: number
}
```

Reply:
I've changed all field name references to use the `enum`.